### PR TITLE
use wrapping arithmetic, increase buffer size to 256

### PR DESCRIPTION
**Pull request by digama0**

### CODE REVIEW

File: src/line_cache.rs [hunk 1]
```diff
@@ -31,18 +32,19 @@ fn make_index(mut buf: &[u8]) -> Vec<u32> {
     while buf.len() >= PAGE {
         let mut page = &buf[0..PAGE];
         buf = &buf[PAGE..];
-        // use an i8 accumulator to maximize the effectiveness of vectorization.
-        // do blocks of 128 because we don't want to overflow the i8.  count
+        // use a u8 accumulator to maximize the effectiveness of vectorization.
+        // do blocks of 256 because we don't want to overflow the u8.  count
```
Comment (sorear) [inline src/line_cache.rs:36]: But what if someone hands you 256 newlines in a row?

Comment (sorear) [inline src/line_cache.rs:36]: `Wrapping` with a page size of 224 is an option, but I feel it's not the best point on the complexity tradeoff; make_index is already a cool 0.3% on the profile when it's used at all.

Comment (digama0) [inline src/line_cache.rs:36]: Not possible with windows newlines anyway (they're spaced out with `\r`). And I believe we assume windows newlines elsewhere in the program (`guess_buffer_name`?).

Comment (sorear) [inline src/line_cache.rs:36]: My copy of `set.mm` uses Unix newlines, and I haven't actually tested it with Windows newlines (although the new world order means you can pretend `\r` is horizontal whitespace and nearly everything will Just Work)

**Pull request closed without merging**